es5-shim.js

```diff
@@ -328,6 +351,99 @@ var strSplit = call.bind(StringPrototype.split);
+var hasToStringTagLegacyArguments = toStr(arguments) !== '[object Arguments]';
+// Just making sure we don't forget to delete this line, unless we can find (
+// and document in a comment) a specific browser that would give true here.
+//var hasToStringTagArrayBug = toStr([]) !== '[object Array]';
+var hasToStringTagArrayBug = false;
+// Just making sure we don't forget to delete this line, unless we can find (
+// and document in a comment) a specific browser that would give true here.
+//var hasToStringTagRegExpBug = toStr(/test/) !== '[object RegExp]';
+var hasToStringTagRegExpBug = false;
+var hasToStringTagInconsistency = hasToStringTagBasicBug ||
+      hasToStringTagLegacyArguments ||
+      hasToStringTagArrayBug ||
+      hasToStringTagRegExpBug;
+
+// Others to check:
+// Safari 8 returns 'object' for typed array constructors - tests added.
+// PhantomJS 1.9 which returns 'function' for `NodeList` instances.
+
+if (hasToStringTagLegacyArguments) {
+  var isDuckTypeArguments = function (value) {
+    return value != null &&
+        typeof value === 'object' &&
+        owns(value, 'length') &&
+        typeof value.length === 'number' &&
+        value.length >= 0 &&
+        owns(value, 'callee') &&
+        isCallable(value.callee) &&
+        !owns(value, 'arguments');
+
+  };
+}
+
+if (hasToStringTagArrayBug) {
+  var isDuckTypeArray = function (value) {
+    return value != null &&
+        typeof value === 'object' &&
+        owns(value, 'length') &&
+        typeof value.length === 'number' &&
+        value.length >= 0 &&
+        !owns(value, 'callee') &&
+        !owns(value, 'arguments');
+  };
+}
+
+if (hasToStringTagRegExpBug) {
+  var isTryCatchRegExp = function (value) {
+    if (value != null) {
+      var type = typeof value;
+      if (type === 'object' || type === 'function') {
+        try {
+          // This should be robust against delete `Function.prototype.call`
+          apply.bind(call, regexExec, [value])();
+          return true;
+        } catch (e) {
+          return false;
+        }
+      }
+    }
+  };
+}
+
+if (hasToStringTagInconsistency) {
+  defineProperties(FunctionPrototype, {
+    call: function () {
+      var thisArg = arguments[0];
+      if (this === to_string) {
+        if (hasToStringTagBasicBug) {
+          if (thisArg === null) {
+            return '[object Null]';
+          }
+          if (typeof thisArg === 'undefined') {
+            return '[object Undefined]';
+          }
+        }
+        if (hasToStringTagLegacyArguments && isDuckTypeArguments(thisArg)) {
+          return '[object Arguments]';
+        }
+        if (hasToStringTagArrayBug && isDuckTypeArray(thisArg)) {
+          return '[object Array]';
+        }
+        if (hasToStringTagRegExpBug && isTryCatchRegExp(thisArg)) {
+          return '[object RegExp]';
+        }
+      }
+      return apply.bind(this, thisArg, baseSlice(arguments, 1))();
```

Comment:
Probably, I will try to make some changes today and see if I can a slot on IE8, perhaps even force the path with a `true` to see what is happening.